core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/internal/CDOBasedRepository.java

```diff
@@ -348,5 +321,102 @@ public void handleCommitInfo(CDOCommitInfo commitInfo) {
 			notification.publish(events());
         }
 	}
+
+	/**
+	 * Verifies if the repository is in a consistent state, and sets the {@link #state repositoryState} accordingly
+	 */
+	@Override
+	public void updateHealth() {
+		Health newHealth = calculateHealth();
+		if (health != newHealth) {
+			LOG.debug("Repository state for {} is changing from {} to {}", getCdoRepository().getRepositoryName(), health, newHealth);
+			health = newHealth;
+		} else {
+			LOG.debug("Repository state for {} is remaining the same: {}", getCdoRepository().getRepositoryName(), health);
+		}
+	}
 	
+	@Override
+	public Health getHealth() {
+		return health;
+	}
+	
+	public Health calculateHealth() {
+		IBranchPath branchPath = BranchPathUtils.createMainPath();
+		List<CDOCommitInfo> cdoCommitInfos = getCDOCommitInfos(branchPath);
+		CommitInfos indexCommitInfos = getIndexCommitInfos(branchPath);
+		
+		boolean emptyDb = cdoCommitInfos.isEmpty();
+		boolean emptyIndex = indexCommitInfos.getItems().isEmpty();
+
+		if (emptyDb && emptyIndex) {
+			return Health.GREEN; // empty dataset
+		}
+		
+		if (emptyDb ^ emptyIndex) {
+			LOG.error("{} is in inconsistent state. CDO {} but INDEX {}", getCdoRepository().getRepositoryName(), contentMessage(emptyDb), contentMessage(emptyIndex));
+			return Health.RED; // either CDO or index was deleted but not the other.
```

Comment:
If the underlying SQL database is empty, then the state should be `Health.RED`, but if there are database commits, but no index commits, then allow developers to reindex the dataset. Use `Health.YELLOW` in this case.